src/com/core/tictactoe/Board.java

```diff
@@ -56,20 +56,20 @@ boolean isNonTaken(String position){
     }
 
     String winnerSign() {
-        for (int[] set : winningPositions) {
-            String matcher = this.valueAtIndex(set[0]);
-            int numberOfMatchesInSet = countMatchesInSet(set, matcher);
-            if (numberOfMatchesInSet == set.length) return matcher;
+        for (int[] winPath : winningPositions) {
+            String currentSign = this.valueAtIndex(winPath[0]);
+            int currentSignsInWinPath = countCurrentSignsInWinPath(winPath, currentSign);
+            if (currentSignsInWinPath == winPath.length) return currentSign;
         }
         return "none";
     }
 
     ArrayList<String> getFreePlaces() {
         String[] initialPlaces = {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
```

Comment:
Similar to another comment, but in this function it might just be better to check that the place isn't "X" or "O", as this will be more extendable.
It would also be good if the function returned a `String[]` type instead of an `ArrayList<String>`.

Reply:
Sam, I'll apply the first part (meaning checking if a place is not "O" or "X"). I would prefer to keep ArrayList as it can adjust to different sizes of array.